modules/kubernetes-addons/opentelemetry-operator/main.tf

```diff
@@ -1,14 +1,288 @@
+  }
+  rule {
+    api_groups = [""]
+    resources  = ["namespaces"]
+    verbs      = ["list", "watch"]
+  }
+  rule {
+    api_groups = [""]
+    resources  = ["serviceaccounts"]
+    verbs      = ["create", "delete", "get", "list", "patch", "update", "watch"]
+  }
+  rule {
+    api_groups = [""]
+    resources  = ["services"]
+    verbs      = ["create", "delete", "get", "list", "patch", "update", "watch"]
+  }
+  rule {
+    api_groups = ["apps"]
+    resources  = ["daemonsets"]
+    verbs      = ["create", "delete", "get", "list", "patch", "update", "watch"]
+  }
+  rule {
+    api_groups = ["apps"]
+    resources  = ["deployments"]
+    verbs      = ["create", "delete", "get", "list", "patch", "update", "watch"]
+  }
+  rule {
+    api_groups = ["apps"]
+    resources  = ["replicasets"]
+    verbs      = ["create", "delete", "get", "list", "patch", "update", "watch"]
+  }
+  rule {
+    api_groups = ["apps"]
+    resources  = ["statefulsets"]
+    verbs      = ["create", "delete", "get", "list", "patch", "update", "watch"]
+  }
+  rule {
+    api_groups = ["autoscaling"]
+    resources  = ["horizontalpodautoscalers"]
+    verbs      = ["create", "delete", "get", "list", "patch", "update", "watch"]
+  }
+  rule {
+    api_groups = ["coordination.k8s.io"]
+    resources  = ["leases"]
+    verbs      = ["create", "get", "list", "update"]
+  }
+  rule {
+    api_groups = ["opentelemetry.io"]
+    resources  = ["opentelemetrycollectors"]
+    verbs      = ["create", "delete", "get", "list", "patch", "update", "watch"]
+  }
+  rule {
+    api_groups = ["opentelemetry.io"]
+    resources  = ["opentelemetrycollectors/finalizers"]
+    verbs      = ["get", "patch", "update"]
+  }
+  rule {
+    api_groups = ["opentelemetry.io"]
+    resources  = ["opentelemetrycollectors/status"]
+    verbs      = ["get", "patch", "update"]
+  }
+  rule {
+    api_groups = ["opentelemetry.io"]
+    resources  = ["instrumentations"]
+    verbs      = ["get", "list", "patch", "update", "watch"]
+  }
+  rule {
+    api_groups = ["authentication.k8s.io"]
+    resources  = ["tokenreviews"]
+    verbs      = ["create"]
+  }
+  rule {
+    api_groups = ["authorization.k8s.io"]
+    resources  = ["subjectaccessreviews"]
+    verbs      = ["create"]
+  }
+}
+
+resource "kubernetes_cluster_role_binding_v1" "adot" {
+  count = var.enable_amazon_eks_adot ? 1 : 0
+
+  metadata {
+    name = local.eks_addon_clusterrole_name
+  }
+  subject {
+    kind      = "User"
+    name      = local.eks_addon_role_name
+    api_group = "rbac.authorization.k8s.io"
+  }
+  role_ref {
+    api_group = "rbac.authorization.k8s.io"
+    kind      = "ClusterRole"
+    name      = local.eks_addon_clusterrole_name
+  }
+
+}
+
+
+module "helm_addon" {
+  source = "../helm-addon"
+  count  = var.enable_opentelemetry_operator ? 1 : 0
+
+  helm_config   = local.helm_config
+  irsa_config   = null
+  addon_context = var.addon_context
+
+  depends_on = [module.cert_manager]
```

Comment:
@bryantbiggs Tried the cluster id output from cert manager, but it doesn't work consistently. For example when the value is already in the tfstate, TF doesn't prioritize cert-manager anymore